### PR TITLE
fix: Tune appPushTimeout capability

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Addresses https://github.com/appium/appium/issues/19990

### CODE REVIEW

File: lib/real-device-management.js [hunk 1]
```diff
@@ -2,12 +2,14 @@ import _ from 'lodash';
...
 
+const DEFAULT_APP_INSTALLATION_TIMEOUT_MS = 8 * 60 * 1000;
+
 /**
  * @typedef {Object} InstallOptions
  *
  * @property {boolean} [skipUninstall] Whether to skip app uninstall before installing it
  * @property {'serial'|'parallel'|'ios-deploy'} [strategy='serial'] One of possible install strategies ('serial', 'parallel', 'ios-deploy')
- * @property {number} [timeout] App install timeout
+ * @property {number} [timeout=600000] App install timeout
```
Comment (KazuCocoa) [inline lib/real-device-management.js:12]:
```suggestion
 * @property {number} [timeout=480000] App install timeout
```
?

File: lib/real-device.js [hunk 2]
```diff
@@ -106,22 +109,27 @@ export class RealDevice {
...
-          timeoutMs: timeout,
-          enableParallelPush:
-            _.toLower(/** @type {'parallel'} */ (strategy)) === APP_INSTALL_STRATEGY.PARALLEL,
-        });
+        await B.resolve(pushFolder(afcService, app, bundlePathOnPhone, {enableParallelPush})).timeout(timeout);
         await this.installOrUpgradeApplication(
           bundlePathOnPhone,
-          await this.isAppInstalled(bundleId),
+          {
+            timeout: Math.max(timeout - timer.getDuration().asMilliSeconds, 60000),
```
Comment (KazuCocoa) [inline lib/real-device.js:120]: does this `60000` want to be `DEFAULT_APP_INSTALLATION_TIMEOUT_MS`? just to check the intention)

File: lib/real-device-management.js (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/real-device-management.js:12]: fixed

File: lib/real-device.js (hunk 2 above)
Comment (mykola-mokhnach) [reply lib/real-device.js:120]: that's ok. this is needed to make sure we at least allow one minute for app installation if files transfer has already almost used the limit

**Review by KazuCocoa — APPROVED** [review]

**Pull request merged**